### ISSUE TITLE
TcBSD : log files cannot be deleted

### ISSUE THREAD

**Issue by KuronekoChris** [bengeisler/TcLog#34]
Hello,

I have a problem deleting logs under TcBSD when the time limit is exceeded.
The function returns error ADS 1804.
![image](https://github.com/user-attachments/assets/c4f33357-5218-4b42-8570-ffa7369c8db7)

The log files are created with the root owner and when we delete them we are unfortunately only Administrator.

![image](https://github.com/user-attachments/assets/33ab1cf5-f8e3-4e75-ab16-b89b44ddc227)

My version of TcBSD : 
![image](https://github.com/user-attachments/assets/24010175-a229-42f6-abc6-158835f7b061)

Is there anything I've done wrong?

Thanks in advance.

KuronekoChris

**Comment by bengeisler** [comment]
Hi,

this is interesting. Unfortunately, I have not yet worked with TcBSD yet and also not tested this library against it. 

Under the hood this library uses `FB_FileWrite` and `FB_FileDelete`, which are standard Beckhoff function blocks. These do not provide inputs for setting any user. I'm wondering why the TwinCAT system would use two different users: one for creating the file (`root`) and one for deleting it (`Administrator`)?

Some questions:
- what is the absolute path of your home directory? Could you run `pwd` here and provide the output?
- What happens if you place the files in `/var/log` instead?
- When you create a test file manually using [FB_FileWrite](https://infosys.beckhoff.com/english.php?content=../content/1033/tcplclib_tc2_system/30974475.html&id=) and delete it with [FB_FileDelete](https://infosys.beckhoff.com/english.php?content=../content/1033/tcplclib_tc2_system/30974475.html&id=), does it work? Which user is shown as owner of that file? Does it vary depending on whether you do it in `~/Logs` or `/var/log`?

**Comment by KuronekoChris** [comment]
Hi,

I tested /var/log/ and it does the same thing.

After several tests, it's probably not the owner's fault. I managed to delete the files created with [FB_FileDelete](https://infosys.beckhoff.com/english.php?content=../content/1033/tcplclib_tc2_system/30974475.html&id=) even though they were root files.

I still don't know why the _fileDelete is telling me that the file cannot be found.

When I'm in debug mode, I can see that the sPathName is correct.
![image](https://github.com/user-attachments/assets/4fa3071f-9bce-4bbd-8d70-f55d85335b65)

![image](https://github.com/user-attachments/assets/5d90a40e-a0e7-4e96-b68a-f05540364eb1)

**Comment by bengeisler** [comment]
This looks like a bug. I won't have access to my system this year anymore, but I'll check this beginning of January. 
Thanks so far for reporting it!

**Comment by seehma** [comment]
this looks a bit like the file is still open. The TwinCAT service  releases all filehandles when it goes into config mode. Can you check if the file can be deleted manually if your plc is in config mode?

**Comment by bengeisler** [comment]
Yes, this sounds reasonable. We had a similar issue [here](https://github.com/bengeisler/TcLog/issues/4). I suspect the state machine responsible for deleting the file has a bug.

**Comment by KuronekoChris** [comment]
Hi,

No problem to delete a Logs file written by TwinCAT. Even in run mode, I can delete the log files manually.
After I set a deletion to 7 days so the files are no longer open normally (?)

**Comment by bengeisler** [comment]
I could reproduce the issue. It should be fixed with the latest release. Could you please confirm that it solves your issue?

Thanks again for filing this bug - and I also found another one: the use of multiple loggers seems to have some interdependency. They don't work in isolation as they should. Will look into this one next.
